### ISSUE TITLE
Reconsider losses, including KL term

### ISSUE THREAD

**Issue by matsen** [matsengrp/vampire#29]
Here's our current loss setup: 
```python
    def vae_loss(io_encoder, io_decoder):
        """
        The loss function is the sum of the cross-entropy and KL divergence.
        """
        # Notice that "objectives.categorical_crossentropy(io_encoder,
        # io_decoder)" is a vector so it is averaged using "K.mean":
        xent_loss = io_decoder.shape.num_elements() * K.mean(
            objectives.categorical_crossentropy(io_encoder, io_decoder))
        kl_loss = -0.5 * K.sum(1 + z_log_var - K.square(z_mean) - K.exp(z_log_var), axis=-1)
        kl_loss *= 1 / 4 * params['batch_size']  # Because we have four input/output pairs.
        return (xent_loss + kl_loss)

    def mean_squared_error_2d(io_encoder, io_decoder):
        def mse(i):
            return keras.losses.mean_squared_error(io_encoder[:, i], io_decoder[:, i])

        return mse(0) + mse(1)
```
and
```python
    vae.compile(
        optimizer="adam",
        loss={
            'cdr3_output': vae_loss,
            'cdr3_length_output': keras.losses.mean_squared_error,
            'v_gene_output': vae_loss,
            'j_gene_output': vae_loss,
            'contiguous_match_output': mean_squared_error_2d
        },
        loss_weights={
            'cdr3_output': 1.,
            'cdr3_length_output': 1000.,
            'v_gene_output': 5.,
            'j_gene_output': 5.,
            'contiguous_match_output': 500.
        })
```

You can see already that this is a little messed up. At the beginning we talk about there being 4 terms, but there's 5 (I forgot to update after adding the last one). But in any case we we have some strange weighting of the KL because of our loss weights.

My proposal is 

1. To only have the KL loss appear in one place (say, the cdr3_output loss)
2. To have an independent weight for it (see https://openreview.net/pdf?id=Sy2fzU9gl and https://arxiv.org/pdf/1602.02282.pdf). We might want this weight to change during training.
